### PR TITLE
Add Collaborations model

### PR DESCRIPTION
**Pull request by jdbohrman**

### CODE REVIEW

File: app/models/collaboration.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#  subject_type      :string
+#  created_at        :datetime         not null
+#  updated_at        :datetime         not null
+#
+class Collaboration < ApplicationRecord
+#	belongs_to :stakeholder, polymorphic: true
+#	belongs_to :subject, polymorphic: true
+
+	# Validations
+	validates :relationship_type, presence: true
```
Comment (StephenForrest) [inline app/models/collaboration.rb:17]: Spacing here is off, should just be 2 spaces/tab

File: db/migrate/20230428215747_create_collaborations.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateCollaborations < ActiveRecord::Migration[7.0]
+  def change
+    create_table :collaborations do |t|
+      t.integer :relationship_type
+      t.string :stakeholder_type
```
Comment (StephenForrest) [inline db/migrate/20230428215747_create_collaborations.rb:5]: It's best to add these as references (as it also indexes them):

```suggestion
      t.references :stakeholder, polymorphic: true
```

File: db/migrate/20230428215747_create_collaborations.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+      t.string :subject_type
```
Comment (StephenForrest) [inline db/migrate/20230428215747_create_collaborations.rb:6]: Same here
```suggestion
      t.references :subject, polymorphic: true
```

File: app/models/collaboration.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+#  relationship_type :integer
+#  stakeholder_type  :string
+#  subject_type      :string
+#  created_at        :datetime         not null
+#  updated_at        :datetime         not null
+#
+class Collaboration < ApplicationRecord
+#	belongs_to :stakeholder, polymorphic: true
+#	belongs_to :subject, polymorphic: true
+
```
Comment (StephenForrest) [inline app/models/collaboration.rb:15]: Somewhere here you want to define the `enum` too:

```suggestion
  enum relationship_type: { stakeholder: 0, owner: 1 }
```

**Review by StephenForrest — CHANGES_REQUESTED** [review]
A few things but overall looks good! Remember BEFORE changing the migration file to `rails db:rollback` to undo the migration before changing it and re-`db:migrate`-ing forwards to change the schema

**Comment by jdbohrman** [comment]
> A few things but overall looks good! Remember BEFORE changing the migration file to `rails db:rollback` to undo the migration before changing it and re-`db:migrate`-ing forwards to change the schema

Should be good to merge now 💪

File: app/models/collaboration.rb (hunk 1 above)
Comment (StephenForrest) [reply app/models/collaboration.rb:24]: Just this spacing but now everything else looks great!

**Review by StephenForrest — APPROVED** [review]

**Pull request merged**